samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -99,13 +104,111 @@ def _get_configuration_address(address: str) -> str:
     Cleans all addresses of indices and returns a clean address
 
     Parameters
-    ==========
+    ----------
     address : str
         The address to clean
 
     Returns
-    =======
+    -------
     str
         The address clean of indices
     """
     return re.sub(r"\[[^\[\]]*\]", "", address)
+
+
+def _resolve_module_output(module: TFModule, output_name: str) -> List[Union[ConstantValue, ResolvedReference]]:
+    """
+    Resolves any references in the output section of the module
+
+    Parameters
+    ----------
+    module : Module
+        The module with outputs to search
+    output_name : str
+        The value to resolve
+
+    Returns
+    -------
+    List[Union[ConstantValue, ResolvedReference]]
+        A list of resolved values
+    """
+    results: List[Union[ConstantValue, ResolvedReference]] = []
+
+    output = module.outputs.get(output_name)
+
+    if not output:
+        raise InvalidResourceLinkingException(f"Output {output_name} was not found in module {module.full_address}")
+
+    output_value = output.value
+
+    LOG.debug("Resolving output {%s} for module {%s}", output_name, module.full_address)
+
+    if isinstance(output, ConstantValue):
+        LOG.debug(
+            "Resolved constant value {%s} for module {%s} for output {%s}",
+            output.value,
+            module.full_address,
+            output_name,
+        )
+
+        results.append(output)
+    elif isinstance(output, References):
+        LOG.debug("Found references for module {%s} for output {%s}", module.full_address, output_name)
+
+        cleaned_references = _clean_references_list(output_value)
+
+        for reference in cleaned_references:
+            if reference.startswith("var."):
+                LOG.debug(
+                    "Resolving variable reference {%s} for module {%s} for output {%s}",
+                    reference,
+                    module.full_address,
+                    output_name,
+                )
+
+                stripped_reference = _get_configuration_address(reference[4:])
```

Comment:
anytime absolute numbers are used one has to be careful, would len("var.") be better?

Also if we are to use multiple if else (in this case its 3 (if, elif, else)), it might be better to use a dispatcher pattern. eg: https://www.oreilly.com/library/view/python-cookbook/0596001673/ch01s07.html

Reply:
Yup, changing the magic number to `len("var.")` will be nicer. Might also consider `reference[reference.find("."):]` to be inline with the other if case that uses `rfind()`

Will look into the dispatcher pattern.

Reply:
So if we were to go with a dispatching design, we would still need some way to handle the `else` case. Since Python doesn't have a switch-like statement until 3.10, we would need to do some checking if `var` or `module` was in the dictionary and throw all the rest of the cases into a default function/branch. From what I can make out, we would still need some `if` or `try/except` to catch a `KeyError` for "default" cases.